source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/UtiliOS.mm

```diff
@@ -278,15 +278,42 @@ void applyBackgroundImageConstraints(const BackgroundImage *backgroundImagePrope
         case ImageFillMode::Cover:
         default: {
             imageView.contentMode = UIViewContentModeScaleAspectFill;
+            // Fill Mode Description
+            // ScaleAspectFill increases one dimension proportionally if
+            // corresponding dimension increases
+            // find which dimension is in deficit and act accordingly
+            // when both dimensions are in deficit find the most deficient dimension
+            // and increase
+            // center the modified image view to the target view.
+
+            CGSize targetViewSize = superView.frame.size;
+            CGSize sourceSize = image.size;
+            BOOL isDeficientInWidth = NO;
+            BOOL isDeficientInHeight = NO;
+
+            if (sourceSize.width < targetViewSize.width) {
+                isDeficientInWidth = YES;
+            }
+
+            if (sourceSize.height < targetViewSize.height) {
+                isDeficientInHeight = YES;
+            }
 
-            if (superView.frame.size.width > imageView.frame.size.width) {
+            if (isDeficientInWidth and isDeficientInWidth) {
```

Comment:
Also, based on my understanding of how UIViewContentModeScaleAspectFill works, it should be sufficient (maybe better) to just set constraints for both the height and width anchors of the view and its superview without checking for the ratios.


`

    [imageView.widthAnchor constraintEqualToAnchor:superView.widthAnchor].active = YES;
    [imageView.heightAnchor constraintEqualToAnchor:superView.heightAnchor].active = YES;
`


instead of:

`

            CGSize targetViewSize = superView.frame.size;
            CGSize sourceSize = image.size;
            BOOL isDeficientInWidth = NO;
            BOOL isDeficientInHeight = NO;

            if (sourceSize.width < targetViewSize.width) {
                isDeficientInWidth = YES;
            }

            if (sourceSize.height < targetViewSize.height) {
                isDeficientInHeight = YES;
            }

            if (isDeficientInWidth and isDeficientInHeight) {
                CGFloat widthDeficiencyRaito = targetViewSize.width / sourceSize.width;
                CGFloat heightDifficiencyRaito = targetViewSize.height / sourceSize.height;
                // we choose one with bigger difficienty in ratio, and by increasing the
                // dimension, the other dimension will be increaed by the same % since it's
                // cover mode
                if (widthDeficiencyRaito >= heightDifficiencyRaito) {
                    [imageView.widthAnchor constraintEqualToAnchor:superView.widthAnchor].active = YES;
                } else {
                    [imageView.heightAnchor constraintEqualToAnchor:superView.heightAnchor].active = YES;
                }
            } else if (isDeficientInWidth) {
                [imageView.widthAnchor constraintEqualToAnchor:superView.widthAnchor].active = YES;
            } else if (isDeficientInHeight) {
                [imageView.heightAnchor constraintEqualToAnchor:superView.heightAnchor].active = YES;
            }

`

Reply:
> Shouldn't one of them be isDeficientInHeight?
> 
> cc- @nesalang @Aditya0603 @shalinijoshi19

that's good find. i don't know how it escaped.

Reply:
Regards to the 2nd point. Your suggestion will affect the dimension of the superview. Try the sample AdaptiveCard.BackgroundImage.FillMode.Cover.json in our samples.

Reply:
@nesalang did you try it with lesser priority as discussed on call?